public/video-ui/src/util/parseGridMetadata.js

```diff
@@ -10,22 +10,69 @@ function parseMimeType(mimeType) {
   return mimeType;
 }
 
-function parseAsset(asset) {
+function parseAsset(asset, aspectRatio) {
   return {
     file: asset.secureUrl,
     mimeType: parseMimeType(asset.mimeType),
     size: asset.size,
+    aspectRatio: aspectRatio,
     dimensions: {
       width: asset.dimensions.width,
-      height: asset.dimensions.height
+      height: asset.dimensions.height,
     }
   };
+};
+
+export function parseImageFromGridCrop(cropData, imageData) {
+  const aspectRatio = cropData.specification.aspectRatio;
+  return {
+    assets: cropData.assets.map((asset) => parseAsset(asset, aspectRatio)),
+    master: parseAsset(cropData.master, aspectRatio),
+    mediaId: cropData.specification.uri,
+    source: imageData.data.metadata.credit
+  };
 }
 
-export function parseImageFromGridCrop(cropData) {
+export function parseComposerDataFromImage(image, trail) {
+
+  const urlParts = image.mediaId.split('/');
+  const mediaId = urlParts[urlParts.length - 1];
+
+  function getComposerAsset(asset) {
+    return {
+      assetType: 'image',
+      mimeType: asset.mimeType,
+      url: asset.file,
+      fields: {
+        width: asset.dimensions.width.toString(),
+        height: asset.dimensions.height.toString(),
+        aspectRatio: asset.aspectRatio
+      }
+    };
+  }
+
+  function getComposerMasterAsset(asset) {
+    const composerAsset = getComposerAsset(asset);
+    composerAsset.isMaster = 'true';
+    return composerAsset;
+  }
+
+  const tempDiv = document.createElement('div');
```

Comment:
This was the best way I found of extracting the text from the trail text (which is html) without using jquery or complicated regexes. Happy to change it if there is saner way of doing it.